pages/products/index.tsx

```diff
@@ -45,15 +53,23 @@ const Products = () => {
 
     return (
         <Panel>
-            <StatefulTable
+            <Table
                 columns={[
-                    { header: 'Product name', hash: 'name', render: ({ id, name }) => renderName(id, name), sortKey: 'name' },
-                    { header: 'Stock', hash: 'stock', render: ({ stock }) => renderStock(stock), sortKey: 'stock' },
-                    { header: 'Price', hash: 'price', render: ({ price }) => renderPrice(price), sortKey: 'price' },
-                    { header: 'Action', hideHeader: true, hash: 'id', render: ({ id }) => renderAction(id), sortKey: 'id' },
+                    { header: 'Product name', hash: 'name', render: ({ id, name }) => renderName(id, name), isSortable: true },
+                    { header: 'Stock', hash: 'stock', render: ({ stock }) => renderStock(stock), isSortable: true },
+                    { header: 'Price', hash: 'price', render: ({ price }) => renderPrice(price), isSortable: true },
+                    { header: 'Action', hideHeader: true, hash: 'id', render: ({ id }) => renderAction(id), isSortable: true },
```

Comment:
in order to properly sort, we need to add an additional top-level, prop to `Table`, `sortable`. We'll also need to add some additional state in order for the Table to properly handle sorting. Please see `Sortable` -https://bigcommerce.github.io/big-design/table
```jsx
const router = useRouter();
const [tableItems, setTableItems] = useState([]);
const [itemsPerPage, setItemsPerPage] = useState(10);
const [currentPage, setCurrentPage] = useState(1);
const [columnHash, setColumnHash] = useState('');
const [direction, setDirection] = useState<'ASC' | 'DESC'>('ASC');
const { error, isLoading, list = [], meta = {} } = useProductList({ page: String(currentPage), limit: String(itemsPerPage) });

useEffect(() => {
    if (list.length) {
        const tableItems: TableItem[] = list.map(({ id, inventory_level: stock, name, price }) => ({
            id,
            name,
            price,
            stock,
        }));
        setTableItems(tableItems);
    }
}, [list]);

const onSort = (newColumnHash: string, newDirection: string) => {
    setColumnHash(newColumnHash);
    setDirection(newDirection);
    setTableItems(currentItems => sort(currentItems, newColumnHash, newDirection));
};

return (
    <Table
        columns={[...]}
        pagination={{ ... }}
        sortable={{
            columnHash,
            direction,
            onSort,
        }}
    />
);
```

Reply:
since the `useEffect` above is passing an array (`list`) as a dependency, we'll want to memoize that array. Most of JS uses referential equality when comparing two objects - since arrays are a type of object, `arr1 === arr2` will always be false, even if they contain exactly the same key/ values. The only time arr1 will equal arr2 is if we explicitly set `arr1 = arr2`.

In our case, this means we'll need to update the hook so that this useEffect doesn't get hit on every lifecycle. In our `lib/hooks` we might be able to do a quick n dirty useMemo as part of the return:
```jsx
return useMemo(() => ({
    list: data?.data,
    meta: data?.meta,
    isLoading: !data && !error,
    error,
    mutateList,
}, [data, error, mutateList]));
```